maat-court-data-api/src/main/java/gov/uk/courtdata/service/QueueMessageLogService.java

```diff
@@ -35,21 +36,41 @@ public void createLog(final MessageType messageType, final String message) {
 
         JsonObject msgObject = JsonParser.parseString(message).getAsJsonObject();
 
-        JsonElement uuid = msgObject.get("laaTransactionId");
```

Comment:
We are missing out the LAA TX ID value in the case of LAA_STATUS_UPDATE. Hence I tried to keep it this way. I'll create a separate story to set up the TX value in the header attribute.

Reply:
Yeah thats a good idea, the suggested logic should only use `metadata.laaTransactionId` if it exists and isn't a REST call otherwise it will fall back on the existing behaviour.

Reply:
I also noticed that for linking and unlink SQS we are getting the laaTransactionId as a part of the main body. Made changes for this and need to update the integration test for both.

Reply:
Do we really need the additional branching though? The `metadata` field is only exists to hold the `laaTransactionId` so surely the logic below covers it? Also I think we just keep the existing section of code to get the `maatId`.

```
JsonElement uuid = msgObject.has("metadata") ?
                msgObject.get("metadata").getAsJsonObject().get("laaTransactionId") :
                msgObject.get("laaTransactionId");
```